File: Laba7/C++/LinkedList.cpp
#include "LinkedList.h"

double LinkedList::MedianValue() {
	double sum = 0.0;
	int count = 0;
	Node* temp = head;
	while (temp != NULL) {
		sum += temp->Data;
		count++;
		temp = temp->Link;
	}
	return sum / count;
}
void LinkedList::DeleteEvenElements() {
	Node* next = head->Link;
	Node* after = head->Link->Link;
	head = next;
	while (next != NULL && after != NULL) {
		next->Link = after->Link;
		after = NULL;
		next = next->Link;
		if (next != NULL) {
			after = next->Link;
		}
	}
}
void LinkedList::PrintList() {
	Node* temp = head;
	while (temp != NULL) {
		cout << temp->Data << "->";
		temp = temp->Link;
	}
	cout << "null" << endl;
}

int LinkedList::FindNumberOfElementsGreaterThanMedian() {
	int count = 0;
	double median = MedianValue();
	Node* temp = head;
	while (temp != NULL) {
		if (temp->Data > median) {
			count++;
		}
		temp = temp->Link;
	}
	return count;
}

File: Laba7/C#/LinkedList.cs
using System;

namespace Laba7
{
    public class LinkedList
    {
        public Node Head;
        
        public class Node
        {
            public double Data { get; }
            public Node Link;

            public Node(double data)
            {
                Data = data;
                Link = null;
                
            }
        }

        public int FindNumberOfElementsGreaterThanMedian()
        {
            var count = 0;
            var median = MedianValue();
            var temp = Head;
            while (temp != null)
            {
                if (temp.Data > median)
                {
                    count++;
                }

                temp = temp.Link;
            }

            return count;
        }

        public void DeleteEvenElements()
        {
            var next = Head.Link;
            var after = Head.Link.Link;
            Head = next;
            while (next != null && after != null)
            {
                next.Link = after.Link;
                after = null;
                next = next.Link;
                if (next != null)
                {
                    after = next.Link;
                }
            }
        }

        public void PrintList()
        {
            var temp = Head;
            while (temp != null)
            {
                Console.Write(temp.Data+ "->");
                temp = temp.Link;
            }

            Console.WriteLine("null");
        }
        private double MedianValue()
        {
            var sum = 0.0;
            var count = 0;
            var temp = Head;
            while (temp != null)
            {
                sum += temp.Data;
                count++;
                temp = temp.Link;
            }

            return sum / count;
        }
    }
}